jcl/src/java.base/share/classes/java/lang/invoke/PrimitiveHandle.java

```diff
@@ -192,7 +192,9 @@ public MethodHandle asVarargsCollector(Class<?> arrayParameter) throws IllegalAr
 			String signature = oldType.toMethodDescriptorString();
 			return lookupMethod(referenceClass, name, signature, kind, specialToken);
 		} catch (NoSuchMethodError e) {
-			throw new NoSuchMethodException(e.getMessage());
+			NoSuchMethodException eWrapper = new NoSuchMethodException();
+			eWrapper.initCause(e);
+			throw eWrapper;
```

Comment:
I'm OK with this change as the NSME already exists

Reply:
updated to exclude the changes to MethodHandles.Lookup.findStatic.